src/ucs/config/parser.c

```diff
@@ -1570,7 +1642,8 @@ ucs_config_parser_print_field(FILE *stream, const void *opts, const char *env_pr
         fprintf(stream, "#\n");
     }
 
-    fprintf(stream, "%s%s%s%s\n", env_prefix, head->prefix, name, value_buf);
+    fprintf(stream, "%s%s%s%s%s\n", sample_ini_prefix, env_prefix, head->prefix,
```

Comment:
Use `#` prefix only when the value is the default. Not if it's modified the environment

Reply:
Determining whether a value is "default" in this context is not straightforward.
Checking `strcmp(&value_buf[1], field->dfl_value)` doesn't make sense, because it might differ even for a default value, e.g `2.0` vs `2.000`, as `sscanf`+`sprintf` in parser are **not** idempotent.

Solutions I can suggest:
1. Check in this context, whether a corresponding env var exists.
2. Running something like scanf(sprintf(dfl_value) and comparing to the actual value.

What do you think? @yosefe

Reply:
IMO, option 2 , more precisely : strcmp(sprintf(value), sprintf(scanf(dfl value))

Reply:
It's trickier than that. Simply doing something like `sprintf(scanf(dfl value))` doesn't make sense - the value need to go through the parser.

Instead, we need to use `parser->read()` then `parser->write()` on the default value, and compare the outcome to the actual field value.

The problem is - we need the size of the field to use these parser funcs, and it is not available in this context.

Do you think allocating on the stack with some const upper bound is reasonable? Like the `char[128]` bufs in this function, but this one will be `void *`.

Reply:
maybe add comparator (is_equal) to each config type?

Reply:
Anyway, there's another issue with this approach.
When there's `super` member in a config struct (e.g `uct_scopy_iface_config_t`), the default values are incorrectly recognized from the base struct.

As a result, in this context we won't comment-out the value, as if it was an env var value, which is wrong.
Maybe we should check for env vars, after all?

Reply:
env var + config file, right?
something like ucs_config_parser_is_default(var_name) which checks if there is env var and/or config file entry for "var_name"?

Reply:
Yes

Reply:
ok